lib/spack/spack/installer.py

```diff
@@ -572,6 +579,21 @@ def log(pkg):
     dump_packages(pkg.spec, packages_dir)
 
 
+def combine_phase_logs(pkg):
```

Comment:
I'm actually ok with not doing this and just shipping separate logs.  We could just make this change for 0.17, though we'd need to modify some of the build and error output to point to the per-phase logs.

@cyrush @white238: are you going to be ok with separate build phase logs?  Or is that going to break uberenv an would you rather have a combined log as well?  This PR allows the logs to be split by phase, and for us to do things like see configure args and other build metadata independently.

If we add nice phase timings for @cyrush does that make you even cooler with this?

Reply:
I added this because I was worried that if a user might depend on that file, removing it would be an issue. I don't think it hurts to keep it.

Reply:
I prefer separate build phase logs.

Reply:
separate build phase logs actually sounds good to me.  
Addition of timestamps would be be awesome as well.

Reply:
@vsoch: let's get rid of the old log given that @cyrush is ok w/this.  Some of the logs can be quite large, so I think it makes sense not to store them twice.

Reply:
Gotcha - will remove! I'm writing the hook functionality now (and also lots of docs) so I will get this in soon (but probably not immediately)

Reply:
@tgamblin removing this is adding a bunch of funky errors - I think it should be done separately from this PR, it's out of scope and can be easily done after.

Reply:
Just to update everyone here - I'm going to leave in the "global log" for now because removing it requires updating several other tests and modules that expect it, and I think the scope of this PR is getting too large (but it will be easy to do, and more cleanly, after in a separate one).

Reply:
Perhaps create an issue referencing this to make sure that doesn't get dropped on the floor?

Reply:
All set: https://github.com/spack/spack/issues/22399 And I'll leave the discussion here un-resolved so if anyone clicks the link to it, the discussion too is not lost in the doldrums of a resolved comment thread.

Reply:
thanks!